embassy-rp/src/spi.rs

```diff
@@ -145,10 +156,19 @@ macro_rules! impl_instance {
 impl_instance!(SPI0, Spi0);
 impl_instance!(SPI1, Spi1);
 
-pub trait ClkPin<T: Instance>: sealed::ClkPin<T> + Pin {}
-pub trait CsPin<T: Instance>: sealed::CsPin<T> + Pin {}
-pub trait MosiPin<T: Instance>: sealed::MosiPin<T> + Pin {}
-pub trait MisoPin<T: Instance>: sealed::MisoPin<T> + Pin {}
+pub trait ClkPin<T: Instance>: sealed::ClkPin<T> + OptionalPin {}
```

Comment:
Should the clock really be optional ?

Reply:
Why not? it's allowed by the hardware. Could be used to output custom waveforms for RGB leds or stuff... except this SPI IP is stupid, it can't transmit bytes back-to-back so timing would be off...